metadata.go

```diff
@@ -0,0 +1,23 @@
+package iavl
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// Marshal returns Protobuf marshaled bytes of a VersionMetadata object.
+func (vm *VersionMetadata) Marshal() ([]byte, error) {
+	return proto.Marshal(vm)
+}
+
+// Unmarshal attempts to decode a VersionMetadata object from Protobuf marshaled
+// bytes.
+func (vm *VersionMetadata) Unmarshal(bz []byte) error {
```

Comment:
Maybe this should be a constructor instead? It's a bit unnecessary to create an empty struct to unmarshal into.